gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/IncrementalTest.kt

```diff
@@ -921,6 +921,73 @@ class IncrementalTest : BaseGradleTest() {
       }
     }
 
+  @TestFactory
+  fun `build cache restores generated source files in a different project location and custom build dir location`() =
+    testFactory {
+
+      // The testKit directory has the daemon and build cache.
+      // We'll use the same testKit directory for both projects
+      // to simulate having a shared remote build cache.
+      val runner = gradleRunner
+        .withTestKitDir(workingDir / "testKit")
+        .withArguments("jar", "--stacktrace")
+
+      val rootA = rootProject(workingDir.resolve("a/root-a")) {}
+      val rootB = rootProject(workingDir.resolve("b/root-b")) {}
+
+      val rootBBuildDir = rootB.path.resolveSibling("root-b-build")
+
+      for (root in listOf(rootA, rootB)) {
+        root.apply {
+          // Copy the normal root project's build and settings files to the new projects.
+          rootProject.buildFileAsFile.copyTo(buildFileAsFile)
+          rootProject.settingsFileAsFile.copyTo(settingsFileAsFile)
+
+          dir("src/main/java") {
+            injectClass()
+          }
+          gradlePropertiesFile(
+            """
+            org.gradle.caching=true
+            com.squareup.anvil.trackSourceFiles=true
+            """.trimIndent(),
+          )
+        }
+      }
+
+      rootB.buildFileAsFile.appendText(
+        """
+          
+        layout.buildDirectory.set(file("../root-b-build"))
+        """.trimIndent(),
+      )
+
+      // Delete the normal auto-generated "root" Gradle files
+      // since we created other projects to work with.
+      rootProject.buildFileAsFile.delete()
+      rootProject.settingsFileAsFile.delete()
+
+      with(runner.withProjectDir(rootA.path).build()) {
+        task(":compileKotlin")?.outcome shouldBe TaskOutcome.SUCCESS
+      }
+
+      rootA.path.deleteRecursivelyOrFail()
```

Comment:
Why do we delete the rootA files here? My initial reaction based on the rest of the setup is that the project A files have been cached, and we're going to verify that those cached files can be used in project B, but that them still existing locally for project A shouldn't affect project B.

Reply:
It's a sanity check, just to ensure that they aren't affecting project B.

Reply:
Followup PR: https://github.com/square/anvil/pull/979#discussion_r1578596878